Sources/SwiftBuildSupport/SwiftBuildSystem.swift

```diff
@@ -585,7 +591,46 @@ public final class SwiftBuildSystem: SPMBuildCore.BuildSystem {
                         throw Diagnostics.fatalError
                     }
 
-                    replArguments = generateReplArguments ? try await self.createREPLArguments(session: session, request: request) : nil
+                    if buildOutputs.contains(.replArguments) {
+                        replArguments = try await self.createREPLArguments(session: session, request: request)
+                    }
+
+                    if buildOutputs.contains(.builtArtifacts) {
+                        if let buildDescriptionID {
+                            let targetInfo = try await session.configuredTargets(buildDescription: buildDescriptionID, buildRequest: request)
+                            artifacts = targetInfo.compactMap { target in
+                                guard let artifactInfo = target.artifactInfo else {
+                                    return nil
+                                }
+                                let kind: PluginInvocationBuildResult.BuiltArtifact.Kind = switch artifactInfo.kind {
+                                case .executable:
+                                    .executable
+                                case .staticLibrary:
+                                    .staticLibrary
+                                case .dynamicLibrary:
+                                    .dynamicLibrary
+                                case .framework:
+                                    // We treat frameworks as dylibs here, but the plugin API should grow to accomodate more product types
+                                    .dynamicLibrary
+                                }
+                                var name = target.name
+                                // FIXME: We need a better way to map between SwiftPM target/product names and PIF target names
+                                if pifTargetName.hasSuffix("-product") {
```

Comment:
This seems like an accident waiting to happen :S

Should we have some more API surface here?